elastic/kibana
# [Uptime] Waterfall Chart - enable timings for local files by leveraging total time

Fixes #86331 

## Summary

This PR creates an entry for a resource in the waterfall chart, even when individual timing values are unavailable, as is the case with local files. We first loop through checking the specific timings. If an entry is not found for specific timing values, in the case where all the specific named timings equal `-1`, then we move on and use the total time.

<img width="1761" alt="Screen Shot 2021-01-11 at 11 47 18 AM" src="https://user-images.githubusercontent.com/11356435/104212044-c7948800-5402-11eb-8074-bdeeefaaff65.png">

Test cases
* Create a synthetic monitor that hits local files on your file system and verify waterfall chart is displaying the timings accurately
* Create a synthetic monitor that hits server hosted content, for regression testing.
